### ISSUE TITLE
App Ideas

### ISSUE THREAD

**Issue by Pertempto** [el-apps/.github#3]
What are some things we might be interested in developing?

We both have had some app ideas floating around in our heads for a while, let's dump them out in the comments.

Maybe we can combine some of the ideas. Maybe some of the ideas are better kept as separate apps that integrate with each other. 🤷🏽

**Comment by Pertempto** [comment]
Some ideas that have been floating around our home - with scattered thoughts related to them. This isn't supposed to be structured, more of a brain dump of ideas. Many of these ideas might be easily combined or modified instead of making separate apps for each.

## Routines App

- daily sequence of alarms and/or notifications
  - SMS?
- toggle off **all** the alarms for a single non-routine day
- vacation mode?

## Reminder App
- send me a text later in the day

## Checklist App
- send a list of checkboxes on a set schedule to a given email address
- the goal is to take the load off of our brains - inspired by the checklists that pilots and astronauts use in emergency situations
- at first the goal would be scheduled checklists, but maybe we could also implement other sorts of triggers?
   - this might be a good place to integrate with other outputs from other apps

## Home Inventory App
- take the benefits of a professional inventory management system and scale it down to be optimized for a home
- support things like
    - quick search for any items
    - optional QR codes for bins/shelves/boxes
    - analytics tools for monitoring things like "when will we run out of bbq sauce?"
    - alerts for when you might be about to run out of important consumables (like bbq sauce :joy:)

## Home Lists App
- a generic app for sharing lists for a family
- basically a notes/to do app, but fancier
- probably very similar to trello
- possible use cases
   - current to do list
   - long term to do/goals list
   - book list
   - want to buy list
- what advantages would a custom app have over just using trello?
    - better support data import/export
    - great integration with our other home apps
        - example - if you are running low on bbq sauce, then the home inventory app could offer the option to add it to your shopping list
- is it really useful to have a generic lists app?
   - it might be more useful to build a generic lists app codebase, then build more specific lists apps off that foundation, each one optimized for a particular use case
   - example optimized list apps
     - shopping list app
         - having the ability to specify quantity quickly and easily would be important
         - integrations with the home inventory and recipe apps would be **super** valuable
     - to do app
         - more focused on making it really easy to check things off a list
     - book list app
         - more focused on things having to do with books themselves, maybe help you quickly find the best price from a certain collection of used book websites
     - goals app
         - more focused on taking big goals and helping you break them down into smaller goals 

## Recipe App
- keep track of a family's favorite recipes
    - it wouldn't really make sense to enter all the recipes in a cookbook into the app, this would be more for the recipes that you use over and over again because your family really likes them 
- easy integration with the home inventory app to check which ingredients are available
- easy integration with the lists app to automatically add ingredients to the shopping list  
- it would be cool if you could easily share recipes with people outside your family


## Meal Planning App
- not sure what all this would include, Ellie does the meal planning in our home :joy:
- this one's value would really come in if it was integrated with the other apps
    - integrate with the recipe and home inventory apps to recommend recipes that could easily be made with supplies on hand
    - integrate with the shopping list app to add items to the shopping list after you've planned the meals you want to make
- really this app could very easily be the same app as the recipe app

## Simple Money Tracking App
- we already do our budgeting/financing in other software, i'm not wanting to implement anything fancy here
- the basic problem
    - we keep track of our budget categories in a monthly spreadsheet. so for example we might budget $500 for food and supplies
    - when we are shopping throughout the month we have no way to know how we're doing relative to the goal
    - it would be great to have a super simple app for
        - monitoring the current amount in a given budget category
        - adding receipts to update the current amounts remaining in each budget category
- working with people's financial data is a real privacy concern - i definitely don't want to store/track their data in any way
   - maybe an integration with Google Drive would be useful here? users could keep their budget data stored in a google sheet, and allow the app access to the sheet?

**Comment by Pertempto** [comment]
Overall, I think we're going to find the most value out of creating super simple apps to start with, that act as basic building blocks.

We'll just get overwhelmed if we try to implement all possible features for an app at once. I think the goal should be creating small blocks of value that we can build on over time. 

For the first iteration of each app, we should think very much with an MVP mindset. The right thing to ask is "what is the minimum viable thing that would have some real value in my day-to-day life?"

**Comment by Pertempto** [comment]
For each app idea we should think carefully before diving in.

- What is the problem we're trying to solve?
- What are the tools on the market that someone could use to solve this problem?
- Why not use the existing tools?
- What benefits would a custom open source app provide that the existing tools do not?
- How often would we actually use the custom app?
- Would the investment of time spent in creating the app be worth the value produced by the app?

**Comment by DavidLandes** [comment]
## Maintenance Log 
### Description
The objective of this app is to record an equipment's information, keep track of related parts, prices, & maintenance tasks.

### Features
- Add equipment that require periodic maintenance.
- Document parts related to specific equipment
- Create maintenance tasks for each part
    - Set up task frequencies. Could integrate with a reminders app or generate Google Calendar or Outlook events

### Possible Equipment
- Cars/Trucks
    - Parts
        - Oil Filter
        - Tires
        - Lights
- Lawn Mower
- HVAC
- Water Softener/Heater/Pump

**Comment by DavidLandes** [comment]
This could get cluttered pretty quick. What if we create a Project called `App Ideas` & create new issues for each app. This organizes discussion to each app's specific issue, while grouping everything together with the project

**Comment by DavidLandes** [comment]
I created https://github.com/orgs/el-apps/projects/1 so we can try it out.

**Comment by Pertempto** [comment]
> This could get cluttered pretty quick. What if we create a Project called App Ideas & create new issues for each app. This organizes discussion to each app's specific issue, while grouping everything together with the project

Good idea! Have a bit more structure will be useful. I really don't mind this issue getting cluttered because I just see it more as a dumping ground for recording stuff that had been scattered on different sticky notes and notebook pages 😆 

Maybe once we have a discord server we could have a channel `app-ideas`
